app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,61 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(_type, _id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      resource_type = data['resource_type']
+      collection_type = resource_type.include?('Host') ? 'hosts' : 'vms'
+
+      # The 'auth_user' param must be deleted since the model will otherwise
+      # pass the data hash directly as params to ConversionHost.new.
+      auth_user = data.delete('auth_user')
+      data['resource'] = resource_search(data['resource_id'], resource_type, collection_class(collection_type))
+
+      ConversionHost.enable_queue(data, auth_user)
```

Comment:
I don't know much about `action_result`, but not here?

Reply:
Near as I could tell, you want to use it in conjunction with api_action. It worked either way, but the logging was better in this case.

Reply:
Although, hm, maybe I should wrap it.

Reply:
@jameswnl Ok, updated it to return an `api_action` result. However, I think I need a 2nd pair of eyes to figure out the test failure. It looks fine, except that for some reason the word "Host" is being left out of the message, and I'm not sure why.

Reply:
@jameswnl Ok, figured it out, the factorybot for hosts doesn't automatically set the type, so I added in the tests. In practice that would always be set for STI.